models/users.js

```diff
@@ -540,6 +545,30 @@ const getUsersBasedOnFilter = async (query) => {
   return [];
 };
 
+const getUsersWithOnboardingState = async (query) => {
+  const time = query.time;
+  const range = Number(time.split("d")[0]);
+  const { allUserStatus } = await getAllUserStatus(query);
+  const allUsersWithOnboardingState = filterUsersWithOnboardingState(allUserStatus);
+  const updatedOnboardingUsersWithDate = [];
+  const filteredUsers = [];
+  for (const user of allUsersWithOnboardingState) {
+    const result = await userModel.doc(user.userId).get();
+    filteredUsers.push(result.data());
+  }
+  filteredUsers.forEach((element) => {
```

Comment:
Nit: instead of `element` we can call `user`.

Reply:
done